### PR TITLE
Cleanups and tests

### PR DESCRIPTION
**Pull request by khaledhosny**

### CODE REVIEW

File: python/psautohint/otfFont.py [hunk 1]
```diff
@@ -341,11 +341,11 @@ def makeHintList(hints, needHintMasks, isH):
             continue
         pos1 = hint[0]
         pos = pos1 - lastPos
-        if isinstance(pos, float) and (int(pos) == pos):
+        if int(pos) == pos:
```
Comment (miguelsousa) [inline python/psautohint/otfFont.py]: would this be a better way to do it?
```python
if not pos % 1:
```
or if you prefer to make it more obvious,
```python
if pos % 1 == 0:
```

Comment (khaledhosny) [reply python/psautohint/otfFont.py]: I considered this, but it did not feel much readable than what you have now, but I can change it if yiu think it is better.

Comment (miguelsousa) [reply python/psautohint/otfFont.py]: I think changing it to one of the above would be better.

Comment (khaledhosny) [reply python/psautohint/otfFont.py]: Done.

**Pull request merged**